cmd/zetae2e/local/bitcoin.go

```diff
@@ -5,71 +5,147 @@ import (
-		startTime := time.Now()
+	// initialize runner for withdraw tests
+	runnerWithdraw := initRunnerWithdraw(conf, deployerRunner, verbose, initNetwork)
+
+	// initialize funds
+	// send BTC to TSS for gas fees and to tester ZEVM address
+	if initNetwork {
+		// mine 101 blocks to ensure the BTC rewards are spendable
+		// Note: the block rewards can be sent to any address in here
+		_, err := runnerDeposit.GenerateToAddressIfLocalBitcoin(101, runnerDeposit.BTCDeployerAddress)
+		require.NoError(runnerDeposit, err)
+
+		// send BTC to ZEVM addresses
+		runnerDeposit.DepositBTC(runnerDeposit.EVMAddress())
+		runnerDeposit.DepositBTC(runnerWithdraw.EVMAddress())
+	}
+
+	// create test routines
+	routineDeposit := createTestRoutine(runnerDeposit, depositTests)
+	routineWithdraw := createTestRoutine(runnerWithdraw, withdrawTests)
 
-		// funding the account
+	// start test routines
+	eg.Go(routineDeposit)
+	eg.Go(routineWithdraw)
+}
+
+// initRunnerDeposit initializes the runner for deposit tests
+func initRunnerDeposit(
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork bool,
+) *runner.E2ERunner {
+	var (
+		name         = "btc_deposit"
+		account      = conf.AdditionalAccounts.UserBitcoin1
+		createWallet = true // deposit tests need Bitcoin node wallet to handle UTXOs
+	)
+
+	return initRunner(name, account, conf, deployerRunner, verbose, initNetwork, createWallet)
+}
+
+// initRunnerWithdraw initializes the runner for withdraw tests
+func initRunnerWithdraw(
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork bool,
+) *runner.E2ERunner {
+	var (
+		name         = "btc_withdraw"
+		account      = conf.AdditionalAccounts.UserBitcoin2
+		createWallet = false // withdraw tests DON'T use Bitcoin node wallet
+	)
+
+	return initRunner(name, account, conf, deployerRunner, verbose, initNetwork, createWallet)
+}
+
+// initRunner initializes the runner for given test name and account
+func initRunner(
+	name string,
+	account config.Account,
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	verbose, initNetwork, createWallet bool,
+) *runner.E2ERunner {
+	// initialize runner for bitcoin test
+	runner, err := initTestRunner(
+		name,
+		conf,
+		deployerRunner,
+		account,
+		runner.NewLogger(verbose, color.FgYellow, name),
+	)
+	testutil.NoError(err)
+
+	// setup TSS address and setup deployer wallet
+	runner.SetupBitcoinAccounts(createWallet)
+
+	// initialize funds
+	if initNetwork {
+		// send some BTC block rewards to the deployer address
+		_, err = runner.GenerateToAddressIfLocalBitcoin(4, runner.BTCDeployerAddress)
+		require.NoError(runner, err)
+
+		// send ERC20 token on EVM
 		txERC20Send := deployerRunner.SendERC20OnEvm(account.EVMAddress(), 1000)
-		bitcoinRunner.WaitForTxReceiptOnEvm(txERC20Send)
+		runner.WaitForTxReceiptOnEvm(txERC20Send)
+
+		// deposit ETH and ERC20 tokens on ZetaChain
+		txEtherDeposit := runner.DepositEther()
+		txERC20Deposit := runner.DepositERC20()
```

Comment:
Do we need ethers and erc20s for the Bitcoin tests?

Reply:
yeah. The `cross-chain swap` test uses it.